src/utils/access_log.ts

```diff
@@ -73,32 +96,83 @@ export const storeSpidLogs = (
     E.map(item => ({
       ...spidLogMsg,
       ...item
-    })),
-    E.fold(
-      err =>
-        TE.left(new Error(`StoreSpidLogs|ERROR=Cannot encrypt payload|${err}`)),
-      (encryptedBlobItem: SpidBlobItem) =>
-        pipe(
-          encryptedBlobItem,
-          t.exact(SpidBlobItem).decode,
-          TE.fromEither,
-          TE.mapLeft(
-            errs =>
-              new Error(
-                `StoreSpidLogs|ERROR=Cannot decode payload|ERROR_DETAILS=${readableReport(
-                  errs
-                )}`
-              )
-          ),
-          TE.chain(spidBlobItem =>
-            upsertBlobFromObject(
-              blobService,
-              containerName,
-              `${spidBlobItem.spidRequestId}-${spidLogMsg.createdAtDay}-${spidLogMsg.fiscalCode}.json`,
-              spidBlobItem
-            )
+    }))
+  );
+};
+
+// Create a writer for azure storage
+export const createAzureStorageAccessLogWriter = (
+  blobService: BlobService,
+  containerName: string
+): AccessLogWriter => (
+  encryptedBlobItem: SpidBlobItem,
+  blobName: string
+): ReturnType<AccessLogWriter> =>
+  pipe(
+    upsertBlobFromObject(
+      blobService,
+      containerName,
+      blobName,
+      encryptedBlobItem
+    ),
+    TE.map(_ => void 0)
+  );
+
+// Create a writer for AWS S3
+export const createAwsS3AccessLogWriter = (
+  s3Service: aws.S3,
+  bucketName: string
+): AccessLogWriter => (
+  encryptedBlobItem: SpidBlobItem,
+  blobName: string
+): ReturnType<AccessLogWriter> =>
+  pipe(
+    TE.tryCatch(
+      () =>
+        new Promise((resolve, reject) =>
+          s3Service.upload(
+            {
+              Body: JSON.stringify(encryptedBlobItem),
+              Bucket: bucketName,
+              Key: blobName
+            },
+            (err, res) => (err ? reject(err) : resolve(res))
           )
-        )
-    )
+        ),
+      E.toError
+    ),
+    TE.map(_ => void 0)
   );
+// Create a writer for a given kind
+export const createAccessLogWriter = (
+  storageKind: AccessLogStorageKind,
+  connectionString: string,
+  containerName: string
+): AccessLogWriter => {
+  if (storageKind === "azurestorage") {
+    return createAzureStorageAccessLogWriter(
+      createBlobService(connectionString),
+      containerName
+    );
+  } else if (storageKind === "awss3") {
+    // In order to keep configurations consistent across different storages,
+    //  we compact parameters into a semicolon-delimited string
+    const [accessKeyId, secretAccessKey, endpoint] = connectionString.split(
+      ";"
+    );
+    return createAwsS3AccessLogWriter(
+      new aws.S3({
+        accessKeyId,
+        endpoint,
+        s3ForcePathStyle: true,
+        secretAccessKey,
+        signatureVersion: "v4"
+      }),
```

Comment:
1. AWS has many ways of getting credentials, forcing to one  (access key + secret key) it might not be convenient for reusability... credential provisioning is handled automatically by the SDK, so the properties aren't required.
2. Unless there are incompatibilities, I would suggest you to upgrade to SDK v3, which allow for an improved modularity and flexibility `@aws-sdk/client-s3`
3. consider endpoint can be `undefined` when using AWS S3


In case might be useful to you, this is an example of S3 file upload via SKD v3 with automatic credential discovery

```js
import * as s3 from "@aws-sdk/client-s3";
const endpoint = "this-is-my-endpoint";
const client = new s3.S3Client({
    endpoint
});
const up = new s3.PutObjectCommand({
    Body: JSON.stringify("encryptedBlobItem"),
    Bucket: "bucketName",
    Key: "blobName"
});
client.send(up);
```